src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/ServiceProviderEngineScope.cs

```diff
@@ -134,9 +138,11 @@ public ValueTask DisposeAsync()
                 }
             }
 
+            ClearState();
```

Comment:
Would it make sense to add this ClearState() call on a finally block to the try-catch block above it?

Reply:
Yes it would.

Reply:
Actually, it's probably better not to try and return this scope to the pool if dispose throws.